extensions/amp-script/0.1/test/unit/test-user-activation-tracker.js

```diff
@@ -73,4 +73,66 @@ describes.realWin('UserActivationTracker', {}, env => {
     expect(tracker.isActive()).to.be.true;
     expect(tracker.getLastActivationTime()).to.equal(ACTIVATION_TIMEOUT + 2);
   });
+
+  describe('expandLongTask', () => {
+    it('should expand when active', () => {
+      const promise = Promise.resolve();
+
+      tracker.activated_({isTrusted: true});
+      tracker.expandLongTask(promise);
+      expect(tracker.isActive()).to.be.true;
+      expect(tracker.isInLongTask()).to.be.true;
+
+      clock.tick(ACTIVATION_TIMEOUT + 1);
+      expect(tracker.isActive()).to.be.true;
+
+      clock.tick(ACTIVATION_TIMEOUT + 1);
+      expect(tracker.isActive()).to.be.true;
+
+      return promise.then(() => {
+        // Skip microtask.
+        return Promise.resolve();
+      }).then(() => {
+        // The gesture window is expanded for an extra window.
+        expect(tracker.isActive()).to.be.true;
+        expect(tracker.isInLongTask()).to.be.false;
+
+        clock.tick(ACTIVATION_TIMEOUT - 1);
+        expect(tracker.isActive()).to.be.true;
+
+        clock.tick(2);
+        expect(tracker.isActive()).to.be.false;
+      });
+    });
+
+    it('should NOT expand when not active', () => {
+      tracker.expandLongTask(Promise.resolve());
+      expect(tracker.isActive()).to.be.false;
```

Comment:
Not sure this assertion fully covers the intended scenario. Maybe we should verify activation time:

```js
const t = tracker.getLastActivationTime();
const promise = Promise.resolve();

expect(tracker.isActive()).to.be.false;
tracker.expandLongTask(promise);
yield promise;

// expandLongTask() is a no-op when inactive.
expect(tracker.getLastActivationTime()).to.equal(t);
expect(tracker.isActive()).to.be.false;
```

Reply:
It should cover it b/c I sleep for a pre-expiration time and check this:

```
clock.tick(ACTIVATION_TIMEOUT - 1);
expect(tracker.isActive()).to.be.true;
```

I.e. this validates that the active status is extended an extra interval.